libmamba/src/core/link.cpp

```diff
@@ -571,7 +571,8 @@ namespace mamba
             fs::create_directories(dst.parent_path());
         }
 
-        if (fs::exists(dst))
+        std::error_code ec;
+        if (fs::exists(dst, ec) && !ec)
```

Comment:
Reminder: @Hind-M appropriately pointed in a meeting to prefer [`mamba::lexists(...)`](https://github.com/mamba-org/mamba/blob/main/libmamba/include/mamba/core/util.hpp#L61-L62)
Although we should still use the error-code version until we add an `expected` version.